### PR TITLE
Add a new test for PredefinedShelfService.save

### LINKED ISSUE #377: Test whether a valid PredefinedShelf can be saved
**Describe the test you would like to add or modify**

Test whether a valid (i.e. non-null) `PredefinedShelf` can be saved. The expectation is that this should pass.

**Additional context**

Method and class that is being tested: e.g. `save()` in `PredefinedShelfService`]

---

If you need anything, feel free to get in touch with us over [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

### PR DESCRIPTION
**Pull request by CGAura**
## Summary of change
Add a test for PredefinedSheflService.save, which checks that a new Predefined shelf can be saved.

## Additional context
This required and updated to the IntegrationTest annotation to ensure that the in memory DB was reset between test files.
As prior to doing this adding a second instance of any PredefinedShelf was breaking other tests, which showed that the tests were not independent. 

## Related issue

Closes #377 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [X] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [X] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [X] Ensure code changes pass all tests

- [X] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [X] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [X] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [X] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [X] Set this pull request to 'draft' if you are still working on it

- [X] Resolved any merge conflicts

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### CODE REVIEW

**Comment by knjk04** [comment]
@CGAura Thanks for working on this! We'll review this shortly

**Comment by CGAura** [comment]
@knjk04 Hi what's the criteria for the hacktoberfest-accepted label

**Comment by knjk04** [comment]
@CGAura PRs that are linked to an issue with the `hacktoberfest` label. We don't actually need the `hacktoberfest-accepted` label on PRs given that our repository has the `hacktoberfest` topic, but since a few people have asked me about this, I've been adding the `hacktoberfest-related` label to PRs for good measure.

I'll add it to this PR as well.

**Comment by CGAura** [comment]
Ah thanks

File: src/test/java/com/karankumar/bookproject/annotations/IntegrationTest.java [hunk 1]
```diff
@@ -36,5 +38,7 @@
 @ExtendWith(SpringExtension.class)
 @SpringBootTest
 @ActiveProfiles("test")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.AUTO_CONFIGURED)
 public @interface IntegrationTest {
 }
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/annotations/IntegrationTest.java:44]: missing newline (as per our style guide)

File: src/test/java/com/karankumar/bookproject/backend/service/PredefinedShelfServiceTest.java [hunk 2]
```diff
@@ -71,4 +74,23 @@ void notSaveNullPredefinedShelf() {
...
+        PredefinedShelf testShelf = new PredefinedShelf(PredefinedShelf.ShelfName.TO_READ);
+        List<PredefinedShelf> expected = Arrays.asList(existingToReadShelf, testShelf);
+
+        // when
+        predefinedShelfService.save(testShelf);
+
+        // then
+        softly.assertThat(predefinedShelfService.count()).isEqualTo(initialCount + 1);
+        softly.assertThat(predefinedShelfService.findAll()).containsAll(expected);
+        softly.assertAll();
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/PredefinedShelfServiceTest.java:94]: assertSoftly can be used here (as per our style guide)

**Review by knjk04 — APPROVED** [review]
Thanks a lot for this, Ali! 

Also, thanks a lot for noticing and fixing the problem of the conflicting tests!

**Comment by knjk04** [comment]
There were a couple of minor style guide violations, but I went ahead and resolved them. I didn't think it was worth another round of code review. Thanks again for your help!

**Comment by CGAura** [comment]
@knjk04 thanks for that.  I'll take look over the style guide again and see if I can spot how I missed those rules.

**Pull request merged**